axelrod/actions.py

```diff
@@ -15,3 +14,13 @@ def flip_action(action: Action) -> Action:
         return Actions.C
     else:
         raise ValueError("Encountered a invalid action.")
+
+
```

Comment:
i made this return a tuple instead of a generator so that it would fail when called, and not fail when the generator reached the offending character.  This makes something like cycler much safer in that it will immediately fail on instantiation and not be a time-bomb waiting to fail when a bad character comes up.